src/core_ocean/shared/mpas_ocn_tidal_potential_forcing.F

```diff
@@ -0,0 +1,665 @@
+   !
+   ! Public member functions
+   !
+   !--------------------------------------------------------------------
+
+   public :: ocn_compute_tidal_potential_forcing, &
+             ocn_tidal_potential_forcing_init
+
+   !--------------------------------------------------------------------
+   !
+   ! Private module variables
+   !
+   !--------------------------------------------------------------------
+
+   logical :: tidalPotentialOn
+   type :: char_array
+     character(:), allocatable :: constituent
+   end type
+   type(char_array), dimension(37) :: constituentList
+
+
+!***********************************************************************
+
+contains
+
+!***********************************************************************
+!
+!  routine ocn_tidal_potential_forcing_layer_thickness
+!
+!> \brief   Computes equilibrium tidal potential 
+!> \author  Steven Brus
+!> \date    September 2019
+!> \details
+!>  This routine computes the equilibrium tidal potential due to all 
+!>  tidal constituents 
+!
+!-----------------------------------------------------------------------
+
+   subroutine ocn_compute_tidal_potential_forcing(meshPool, forcingPool, diagnosticsPool, err)!{{{
+
+      !-----------------------------------------------------------------
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+      type (mpas_pool_type), intent(in) :: meshPool !< Input: mesh information
+      type (mpas_pool_type), intent(in) :: diagnosticsPool 
+
+      !-----------------------------------------------------------------
+      !
+      ! input/output variables
+      !
+      !-----------------------------------------------------------------
+      type (mpas_pool_type), intent(inout) :: forcingPool 
+
+      !-----------------------------------------------------------------
+      !
+      ! output variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, intent(out) :: err !< Output: Error flag
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, pointer :: nCells
+      real (kind=RKIND), pointer :: tidalPotentialRamp
+      real (kind=RKIND), dimension(:), pointer :: lonCell
+      integer, dimension(:), pointer ::  maxLevelCell
+
+      integer, pointer :: nTidalConstituents
+      real (kind=RKIND), dimension(:), pointer :: amplitude, frequency, loveNumbers
+      real (kind=RKIND), dimension(:), pointer :: nodalFactorAmplitude, nodalFactorPhase, astronomicalArgument
+      real (kind=RKIND), dimension(:,:), pointer :: latitudeFunction
+      integer, dimension(:), pointer :: constituentType
+      real (kind=RKIND), dimension(:), pointer :: eta
+      real (kind=RKIND), dimension(:,:), pointer :: zMid, tidalPotentialZMid
```

Comment:
We don't need these here, do we?

Reply:
cc @sbrus89

Reply:
Correct, the `zMid` and `tidalPotentialZMid` could be removed. At one point I was using them here, but moved those calculations elsewhere and forgot to get rid of the declarations. Thanks for catching this.